src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -62,12 +62,12 @@ public override void VisitMarkupTextLiteral(MarkupTextLiteralSyntax node)
 
     public override void VisitMarkupLiteralAttributeValue(MarkupLiteralAttributeValueSyntax node)
     {
-        AddSemanticRange(node, _razorSemanticTokensLegend.MarkupAttributeValue);
+        AddSemanticRange(node, _razorSemanticTokensLegendService.TokenTypes.MarkupAttributeValue);
     }
 
     public override void VisitMarkupAttributeBlock(MarkupAttributeBlockSyntax node)
     {
-        var legend = _razorSemanticTokensLegend;
+        var legend = _razorSemanticTokensLegendService.TokenTypes;
```

Comment:
nit: would it make sense to rename this to tokenTypes

Reply:
It would.. and you know what, this class name is wrong too, it is not specific to TagHelpers at all. Renames incoming!